src/Controllers/SchemaController.js

```diff
@@ -549,56 +549,47 @@ export default class SchemaController {
   }
 
   reloadData(options: LoadSchemaOptions = { clearCache: false }): Promise<any> {
-    let promise = Promise.resolve();
-    if (options.clearCache) {
-      promise = promise.then(() => {
-        return this._cache.clear();
-      });
-    }
     if (this.reloadDataPromise && !options.clearCache) {
       return this.reloadDataPromise;
     }
-    this.reloadDataPromise = promise
-      .then(() => {
-        return this.getAllClasses(options).then(
-          allSchemas => {
-            this.schemaData = new SchemaData(allSchemas, this.protectedFields);
-            delete this.reloadDataPromise;
-          },
-          err => {
-            this.schemaData = new SchemaData();
-            delete this.reloadDataPromise;
-            throw err;
-          }
-        );
-      })
+    this.reloadDataPromise = this.getAllClasses(options)
+      .then(
+        allSchemas => {
+          this.schemaData = new SchemaData(allSchemas, this.protectedFields);
+          delete this.reloadDataPromise;
+        },
+        err => {
+          this.schemaData = new SchemaData();
+          delete this.reloadDataPromise;
+          throw err;
+        }
+      )
       .then(() => {});
     return this.reloadDataPromise;
   }
 
   getAllClasses(
     options: LoadSchemaOptions = { clearCache: false }
   ): Promise<Array<Schema>> {
-    let promise = Promise.resolve();
     if (options.clearCache) {
-      promise = this._cache.clear();
+      return this.setAllClasses();
     }
-    return promise
-      .then(() => {
-        return this._cache.getAllClasses();
-      })
-      .then(allClasses => {
-        if (allClasses && allClasses.length && !options.clearCache) {
-          return Promise.resolve(allClasses);
-        }
-        return this._dbAdapter
-          .getAllClasses()
-          .then(allSchemas => allSchemas.map(injectDefaultSchema))
-          .then(allSchemas => {
-            return this._cache.setAllClasses(allSchemas).then(() => {
-              return allSchemas;
-            });
-          });
+    return this._cache.getAllClasses().then(allClasses => {
+      if (allClasses && allClasses.length) {
+        return Promise.resolve(allClasses);
+      }
+      return this.setAllClasses();
+    });
+  }
+
+  setAllClasses(): Promise<Array<Schema>> {
+    return this._dbAdapter
+      .getAllClasses()
+      .then(allSchemas => allSchemas.map(injectDefaultSchema))
+      .then(allSchemas => {
```

Comment:
It is a small improvement, but I think we don't need to wait the cache promise to resolve. I'd do something like this::
```
this._cache.setAllClasses(allSchemas).catch(error => console.error('Error saving schema to cache', error));
return allSchemas;
```

Reply:
I agree